### PR TITLE
refactor: architecture documentation

### PR DESCRIPTION
**Pull request by raulb**
### Description

- Leaves https://conduit.io/docs/getting-started/architecture as the main source of truth when it comes architecture information, referencing that one from the README.
- Creates a separate page for the pkg structure to not lose that information that's relevant to this repository.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: docs/package_structure.md [hunk 1]
```diff
@@ -5,29 +5,30 @@
...
       this code will be extracted into separate repositories, what will be left is a plugin service that manages built-in
       and external plugins.
-    - `processor` - Provides the types for processing a `Record`. A common abbreviation for transforms is `txf`.
-        - `transform/txfbuiltin` - Contains built-in transforms.
-        - `transform/txfjs` - Provides the functionality for implementing a transform in JavaScript.
-    - `record` - Everything regarding a `Record`, that is the central entity that is pushed through a Conduit pipeline.
+  - `processor` - Provides the types for processing a `Record`. A common abbreviation for transforms is `txf`.
+    - `transform/txfbuiltin` - Contains built-in transforms.
+    - `transform/txfjs` - Provides the functionality for implementing a transform in JavaScript.
+  - `record` - Everything regarding a `Record`, that is the central entity that is pushed through a Conduit pipeline.
```
Comment (hariso) [inline docs/package_structure.md:29]: Some info here is outdated... E.g. there's not more database package (moved to commons), also this transform stuff is outdated too. I'd say it's OK if we do it another PR.

**Comment by hariso** [comment]
Thanks for this @raulb ! WDYT about leaving a link to the architecture page somewhere in this repo (e.g. README)?

**Comment by raulb** [comment]
> Thanks for this @raulb ! WDYT about leaving a link to the architecture page somewhere in this repo (e.g. README)?

@hariso It's there! https://github.com/ConduitIO/conduit/pull/1732/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R293

File: docs/package_structure.md (hunk 1 above)
Comment (raulb) [reply docs/package_structure.md:29]: @hariso I can address it now

File: README.md [hunk 2]
```diff
@@ -290,10 +290,12 @@ documentation:
   explains how you can define pipelines using YAML files.
```
Comment (lovromazgon) [inline README.md:290]: Can we also change the first two links here? I think we have equivalent pages on conduit.io. And if there are no substantial differences, we could even delete the docs from the repo.

Comment (raulb) [reply README.md:290]: @lovromazgon i thought about doing that, but then I felt I was scope creeping a bit. I'll do a quick pass

File: docs/package_structure.md (hunk 1 above)
Comment (raulb) [reply docs/package_structure.md]: @hariso wdyt? https://github.com/ConduitIO/conduit/pull/1732/commits/41dd15a4f65aaed13b641e500f9952a228ef2b37

Comment (hariso) [reply docs/package_structure.md]: Great work!

**Comment by hariso** [comment]
> > Thanks for this @raulb ! WDYT about leaving a link to the architecture page somewhere in this repo (e.g. README)?
> 
> @hariso It's there! https://github.com/ConduitIO/conduit/pull/1732/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R293

Sorry, missed it!

**Review by hariso — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (raulb) [reply README.md:321]: @lovromazgon done https://github.com/ConduitIO/conduit/pull/1732/commits/c524614621d425fdc6bd291813486af42e5a1012. Thought it needed more work.
